### PR TITLE
Borrower related commands

### PR DESCRIPTION
**Pull request by linyutinglyt**
Updated RegisterCommandParser and Tests, and also ServeCommand and its Tests

### CODE REVIEW

File: src/main/java/seedu/address/model/ModelManager.java [hunk 1]
```diff
@@ -234,18 +231,13 @@ public ReadOnlyBorrowerRecords getBorrowerRecords() {
 
 
     @Override
-    public Optional<Borrower> getServingBorrower() {
-        // TODO
-        // return servingBorrower;
-        return Optional.of(new Borrower(new Name("Stub"), new Phone("12345"), new Email("mail@fakemail.co"),
-                new BorrowerId("K1234"))); // STUB
+    public Borrower getServingBorrower() {
+        return servingBorrower.get();
```
Comment (seeziyang) [inline src/main/java/seedu/address/model/ModelManager.java:238]: Maybe we can add an assertion here that servingBorrower isPresent()

File: src/main/java/seedu/address/logic/commands/ServeCommand.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.borrower.Borrower;
+import seedu.address.model.borrower.BorrowerId;
+
+/**
+ * Opens a serving session for a borrower and allows the borrower to start borrower book
+ */
+public class ServeCommand extends Command {
```
Comment (seeziyang) [inline src/main/java/seedu/address/logic/commands/ServeCommand.java:14]: Have you added ServeCommandParser and updated the CatalogParser?

File: src/main/java/seedu/address/commons/core/Messages.java [hunk 3]
```diff
@@ -14,6 +14,7 @@
     public static final String MESSAGE_INVALID_SERIAL_NUMBER = "Invalid Serial Number! \n%1$s";
 
     public static final String MESSAGE_DUPLICATE_BORROWER = "Phone/Email is already in used! ";
+    public static final String MESSAGE_NO_SUCH_BORROWERID = "No such borrower ID!";
```
Comment (seeziyang) [inline src/main/java/seedu/address/commons/core/Messages.java]: I think there should be an underscore between "BORROWER" and "ID", i.e., it should be MESSAGE_NO_SUCH_BORROWER_ID

File: src/main/java/seedu/address/logic/parser/RegisterCommandParser.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+ * Parses input arguments and creates a new RegisterCommand object
+ */
+public class RegisterCommandParser implements Parser<RegisterCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the RegisterCommand
+     * and returns an RegisterCommand object for execution.
+     *
+     * @return RegisterCommand object.
+     * @throws seedu.address.logic.parser.exceptions.ParseException if the user input does not conform the
```
Comment (seeziyang) [inline src/main/java/seedu/address/logic/parser/RegisterCommandParser.java]: Should this just be @throws ParseException

**Review by seeziyang — COMMENTED** [review]
Looks good other than a few small changes!

**Review by hoholyin — COMMENTED** [review]
Looks good!

File: src/main/java/seedu/address/model/ModelManager.java (hunk 1 above)
Comment (linyutinglyt) [reply src/main/java/seedu/address/model/ModelManager.java:238]: Good point! Will add on!

File: src/main/java/seedu/address/logic/commands/ServeCommand.java (hunk 2 above)
Comment (linyutinglyt) [reply src/main/java/seedu/address/logic/commands/ServeCommand.java:14]: Not yet! Will include in next PR. This PR is to assist holyin in his implementations.

File: src/main/java/seedu/address/commons/core/Messages.java (hunk 3 above)
Comment (linyutinglyt) [reply src/main/java/seedu/address/commons/core/Messages.java]: I agree! Will make the necessary amendments !

File: src/main/java/seedu/address/logic/parser/RegisterCommandParser.java (hunk 4 above)
Comment (linyutinglyt) [reply src/main/java/seedu/address/logic/parser/RegisterCommandParser.java]: Yes it should!

**Pull request merged**